pkg/apis/org/v1/che_types.go

```diff
@@ -414,7 +414,7 @@ type CheClusterSpecK8SOnly struct {
 }
 
 type CheClusterSpecMetrics struct {
-	// Enables `metrics` Che server endpoint. Default to `false`.
+	// Enables `metrics` Che server endpoint. Default to `true`.
```

Comment:
Sorry to come too late here, but I'm not sure this documentation change is correct.
The value you propose in the CSV `ALMExamples` field (that will then be provided by `OperatorHub` when clicking on `Create CheCluster` button *is not the same thing as the __default__ field value*

The real __default__ value is the value that will be seen by the Che operator controller when no `metrics.enable` field is set. And since this `enable` field is a GO boolean, according to typical GO json serialization rules, omitting it is the same as setting its value to `false`. So I assume that default value is still `false` in this sense.

Strictly speaking, if we want to always enable metrics by default even when a user provides its own `CheCluster` custom resource without the `metrics.enable` field, then we should probably either:
1. Deprecate the `enable` field and add a new `disable` boolean field that would be `false` by default (typical GO behavior)
2. Have the `enable` field be a pointer to a boolean, so that you can distinguish between the *no value* and *false* cases. However this would not be backward-compatible with the previous versions of the CRD, so I'd rather go for option 1.

I assume this can be done in a future PR.

cc @ibuziuk @tolusha @l0rd

Reply:
@davidfestal 
I see you point. 
What about setting it to `true` if `metrics.enable` field isn't explicitly defined in CR?

Reply:
why, should we do this can't we simply fal back on default CR as described in the docs PR - https://github.com/eclipse/che-docs/pull/1366

Reply:
verified it manually and this works just fine - https://github.com/eclipse/che-operator/pull/322#issuecomment-653566390

Reply:
I see.
`chectl` uses the `deploy/crds/org_v1_che_cr.yaml`  CR. So, metrics will be enabled.

Reply:
> What about setting it to `true` if `metrics.enable` field isn't explicitly defined in CR?

if `metrics.enable` field isn't explicitly defined in CR, it is seen as `false` by the controller afaict. When not using a pointer you cannot distinguish between 'no explicitly defined' and 'defined to false'.